pilot/pkg/networking/util/util.go

```diff
@@ -520,3 +521,23 @@ func MergeAnyWithAny(dst *types.Any, src *types.Any) (*types.Any, error) {
 
 	return retVal, nil
 }
+
+// logPanic logs the caller tree when a panic occurs.
+func logPanic(r interface{}) {
+	// Same as stdlib http server code. Manually allocate stack trace buffer size
+	// to prevent excessively large logs
+	const size = 64 << 10
+	stacktrace := make([]byte, size)
+	stacktrace = stacktrace[:runtime.Stack(stacktrace, false)]
+	log.Errorf("Observed a panic: %#v (%v)\n%s", r, r, stacktrace)
+}
+
+// HandleCrash catches the crash and calls additional handlers.
+func HandleCrash(handlers ...func()) {
```

Comment:
nit: this whole util package is an anti-pattern. Suggest creating a dedicated package for panic handling if you want to share it.

Reply:
+1